### PR TITLE
bsp: imx: imx93-11x11-lpddr4x-evk: add support for i.MX 93 EVK

### PR DESCRIPTION
**Pull request by angolini**
Add the initial support as from the LF5.15.71_2.2.0 NXP BSP release.

### CODE REVIEW

**Comment by angolini** [comment]
It's the initial config to get the yocto building. I can work to reduce the main file tomorrow.

**Review by MrCry0 — APPROVED** [review]
LGTM

**Comment by ricardosalveti** [comment]
Yeah, good as a first reference but it would indeed be good to do a bit of clean up on the cfg file itself, as it is duplicating quite a bit (and enabling more stuff we use/need).

**Comment by angolini** [comment]
I reduced the duplication, if there is still something, please let me know.

I have removed the imx8* references, it passes a simple boot test.

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 1]
```diff
@@ -0,0 +1,903 @@
+# CONFIG_COMPAT_BRK is not set
+CONFIG_ARCH_MXC=y
+CONFIG_ARCH_S32=y
+CONFIG_SOC_S32V234=y
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:4]: Is S32 needed?

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 2]
```diff
@@ -0,0 +1,903 @@
...
+CONFIG_ARM64_VA_BITS_48=y
+CONFIG_SCHED_MC=y
+CONFIG_SCHED_SMT=y
+CONFIG_NUMA=y
+CONFIG_KEXEC=y
+CONFIG_KEXEC_FILE=y
+CONFIG_CRASH_DUMP=y
+CONFIG_XEN=y
+CONFIG_FORCE_MAX_ZONEORDER=14
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:13]: NUMA, KEXEC, CRASH_DUMP and XEN can be removed.

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 3]
```diff
@@ -0,0 +1,903 @@
...
+CONFIG_QCOM_SCM=m
+CONFIG_EFI_CAPSULE_LOADER=y
+CONFIG_IMX_DSP=y
+CONFIG_IMX_SCU=y
+CONFIG_IMX_SCU_PD=y
+CONFIG_ACPI=y
+CONFIG_ACPI_APEI=y
+CONFIG_ACPI_APEI_GHES=y
+CONFIG_ACPI_APEI_MEMORY_FAILURE=y
+CONFIG_ACPI_APEI_EINJ=y
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:46]: Safe do drop ACPI, we don't use it.

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 4]
```diff
@@ -0,0 +1,903 @@
...
+# CONFIG_CORE_DUMP_DEFAULT_ELF_HEADERS is not set
+CONFIG_KSM=y
+CONFIG_MEMORY_FAILURE=y
+CONFIG_TRANSPARENT_HUGEPAGE=y
+CONFIG_NET=y
+CONFIG_PACKET=y
+CONFIG_UNIX=y
+CONFIG_INET=y
+CONFIG_NF_CONNTRACK=m
+CONFIG_NF_CONNTRACK_EVENTS=y
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:71]: NF_ can be removed.

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 5]
```diff
@@ -0,0 +1,903 @@
...
+CONFIG_SRAM=y
+CONFIG_PCI_ENDPOINT_TEST=y
+CONFIG_EEPROM_AT24=m
+CONFIG_EEPROM_AT25=m
+CONFIG_UACCE=m
+# CONFIG_SCSI_PROC_FS is not set
+CONFIG_BLK_DEV_SD=y
+CONFIG_SCSI_SAS_ATA=y
+CONFIG_SCSI_HISI_SAS=y
+CONFIG_SCSI_HISI_SAS_PCI=y
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:141]: HISI can be removed.

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 6]
```diff
@@ -0,0 +1,903 @@
...
+CONFIG_USB_NET_CDC_NCM=m
+CONFIG_USB_NET_DM9601=m
+CONFIG_USB_NET_SR9800=m
+CONFIG_USB_NET_SMSC75XX=m
+CONFIG_USB_NET_SMSC95XX=m
+CONFIG_USB_NET_NET1080=m
+CONFIG_USB_NET_PLUSB=m
+CONFIG_USB_NET_MCS7830=m
+CONFIG_USB_NET_CDC_SUBSET=m
+CONFIG_USB_NET_ZAURUS=m
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:241]: You can drop all usb net adapters.

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 7]
```diff
@@ -0,0 +1,903 @@
...
+CONFIG_NVMEM_IMX_OCOTP=y
+CONFIG_NVMEM_IMX_OCOTP_SCU=y
+CONFIG_NVMEM_RMEM=m
+CONFIG_FPGA=y
+CONFIG_FPGA_BRIDGE=m
+CONFIG_ALTERA_FREEZE_BRIDGE=m
+CONFIG_FPGA_REGION=m
+CONFIG_OF_FPGA_REGION=m
+CONFIG_TEE=y
+CONFIG_OPTEE=y
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:798]: TEE/OPTEE can be removed, they are provided by tee.cfg.

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 8]
```diff
@@ -0,0 +1,903 @@
...
+CONFIG_MUX_MMIO=y
+CONFIG_SLIM_QCOM_CTRL=m
+CONFIG_MXC_SIM=y
+CONFIG_MXC_EMVSIM=y
+CONFIG_EXT2_FS=y
+CONFIG_EXT3_FS=y
+CONFIG_EXT4_FS_POSIX_ACL=y
+CONFIG_BTRFS_FS=m
+CONFIG_BTRFS_FS_POSIX_ACL=y
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:807]: EXT2/4/BTRFS can be removed, they are added via fragments.

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 9]
```diff
@@ -0,0 +1,903 @@
...
+CONFIG_CUSE=m
+CONFIG_OVERLAY_FS=m
+CONFIG_VFAT_FS=y
+CONFIG_TMPFS_POSIX_ACL=y
+CONFIG_HUGETLBFS=y
+CONFIG_EFIVAR_FS=y
+CONFIG_JFFS2_FS=y
+CONFIG_UBIFS_FS=y
+CONFIG_SQUASHFS=y
+CONFIG_SQUASHFS_XZ=y
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:822]: JFFS2/UBI/Squash can be removed.

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 10]
```diff
@@ -0,0 +1,903 @@
...
+CONFIG_NFS_FS=y
+CONFIG_NFS_V4=y
+CONFIG_NFS_V4_1=y
+CONFIG_NFS_V4_2=y
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:826]: Same for NFS.

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 11]
```diff
@@ -0,0 +1,903 @@
...
+CONFIG_DEBUG_INFO_REDUCED=y
+CONFIG_MAGIC_SYSRQ=y
+CONFIG_DEBUG_FS=y
+CONFIG_DEBUG_KERNEL=y
+# CONFIG_SCHED_DEBUG is not set
+# CONFIG_DEBUG_PREEMPT is not set
+# CONFIG_FTRACE is not set
+CONFIG_CORESIGHT=y
+CONFIG_CORESIGHT_LINK_AND_SINK_TMC=y
+CONFIG_CORESIGHT_SOURCE_ETM4X=y
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:891]: Is CORESIGHT used at all?

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 12]
```diff
@@ -0,0 +1,903 @@
...
+CONFIG_MEMTEST=y
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:892]: Can be removed.

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 13]
```diff
@@ -0,0 +1,903 @@
...
+CONFIG_TLS=y
+CONFIG_TLS_DEVICE=y
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:894]: Can be removed, from fragment.

File: bsp/imx/imx93-11x11-lpddr4x-evk.cfg [hunk 14]
```diff
@@ -0,0 +1,903 @@
...
+
+# enable Trusted Keys based on CAAM
+CONFIG_TRUSTED_KEYS=m
+CONFIG_TRUSTED_KEYS_TPM=n
+CONFIG_TRUSTED_KEYS_TEE=n
+CONFIG_TRUSTED_KEYS_CAAM=y
+
+CONFIG_SOC_IMX9=y
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.cfg:902]: Move to the top.

File: bsp/imx/imx93-11x11-lpddr4x-evk.scc [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+include features/pwm/pwm.scc
+include features/crypto/arm64-crypto.scc
+include features/power/cpu-freq.scc
+include features/tee/tee.scc
+
+include cfg/usb-mass-storage.scc
+
+# Exclude x86/ACPI features
+include cfg/non-x86.cfg
+include cfg/fs/ext4.cfg
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk.scc:51]: ext4 comes from standard, can be removed.

File: bsp/imx/imx93-11x11-lpddr4x-evk-preempt-rt.scc [hunk 16]
```diff
@@ -0,0 +1,9 @@
+define KMACHINE imx93-11x11-lpddr4x-evk
+define KARCH aarch64
+define KTYPE preempt-rt
+
+include ktypes/standard/standard.scc
+
+include imx93-11x11-lpddr4x-evk.scc
+
+include ktypes/preempt-rt/preempt-rt.scc
```
Comment (ricardosalveti) [inline bsp/imx/imx93-11x11-lpddr4x-evk-preempt-rt.scc:9]: We should probably drop preempt from this commit, as it needs to be tested and so on first.

**Review by ricardosalveti — APPROVED** [review]
LGTM, a few minor options can be removed but I will follow on that with another pr as I will have to test on hw.

**Pull request merged**